### DISCUSSION TITLE
What is Secret Source and why is set to "None" in my action?

### DISCUSSION THREAD

**Discussion by manast** [actions/runner#3039] [Q&A]
Yo!

I have been spending endless hours trying to get secrets to work on one of my flows without success. On the other flow in the same repo it works. Finally I manage to find something that seems to be the culprit, however there does not seem to exist any documentation about it, namely how are you supposed to configure "secret source"? on one of the flows is set to "Actions", on the other is set to "None". I cannot find anywere in the yaml file where this is supposed to be configured nor is it mentioned on the documentation as far as I can see... Please advise.

On this action it is correctly set to Actions:
https://github.com/taskforcesh/bullmq/actions/runs/7189847132/job/19582017047
![image](https://github.com/actions/runner/assets/95200/712115f3-5ca9-4a3c-a279-7c7cf696332f)

On this other action (note, same exact repo):
https://github.com/taskforcesh/bullmq/actions/runs/7198372591/job/19607650427
![Uploading image.png…]()

What gives?

**Comment by manast** [comment]
Ok. the issue is that PRs do not allow secrets. This should probably be made much more explicit, for instance if I use secrets in a PR action give me a warning or something instead of just silently putting an empty string, would save a lot of time for the next victim of this.

**Comment by papa99do** [comment]
Hello I've encountered the same issue. I don't think my issue is due to PR. PRs raised by code owners or the anyone belonging to the org account can run without any issue, and the secret source is `Action`. For PRs raised by community users, the secret source is `None`. See an example here: https://github.com/marqo-ai/marqo/actions/runs/11115881449/job/30888792261. Is there a way to solve this issue? Thanks
BTW, I'm one of the code owners, I manually retriggered the build and still run into the same issue.

**Comment by bjhargrave** [comment]
From https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflows-in-forked-repositories:

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
